engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -1248,6 +1248,9 @@ public int hashCode()
   @Override
   public <T extends Operator> T addOperator(String name, T operator)
   {
+    if (name == null || name.isEmpty()) {
+      throw new IllegalArgumentException("name argument is null or empty");
```

Comment:
Be consistent with other `IllegalArgumentException`: it should either be "operator id is null or empty" or change "id" to "name" in other places too (I would prefer changing "id" to "name").